sdk/core/azure-core/src/http/winhttp/win_http_transport.cpp

```diff
@@ -4,18 +4,630 @@
+  auto statusLineEnd = std::find(start, last, '\0');
+  start = statusLineEnd + 1; // start of headers
+  std::string responseHeaders = WideStringToString(std::wstring(start, last));
+
+  DWORD sizeOfHttp = sizeOfHeaders;
+
+  // Get the HTTP version.
+  if (!WinHttpQueryHeaders(
+          handleManager->m_requestHandle,
+          WINHTTP_QUERY_VERSION,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          outputBuffer.data(),
+          &sizeOfHttp,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    GetErrorAndThrow("Error while querying response headers.");
+  }
+
+  start = outputBuffer.begin();
+  // Assuming ASCII here is OK since the input is expected to be an HTTP version string.
+  std::string httpVersion = WideStringToStringASCII(start, start + sizeOfHttp / sizeof(WCHAR));
+
+  uint16_t majorVersion = 0;
+  uint16_t minorVersion = 0;
+  ParseHttpVersion(httpVersion, &majorVersion, &minorVersion);
+
+  DWORD statusCode = 0;
+  DWORD dwSize = sizeof(statusCode);
+
+  // Get the status code as a number.
+  if (!WinHttpQueryHeaders(
+          handleManager->m_requestHandle,
+          WINHTTP_QUERY_STATUS_CODE | WINHTTP_QUERY_FLAG_NUMBER,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          &statusCode,
+          &dwSize,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    GetErrorAndThrow("Error while querying response headers.");
+  }
+
+  HttpStatusCode httpStatusCode = static_cast<HttpStatusCode>(statusCode);
+
+  // Get the optional reason phrase.
+  std::string reasonPhrase;
+  DWORD sizeOfReasonPhrase = sizeOfHeaders;
+
+  if (WinHttpQueryHeaders(
+          handleManager->m_requestHandle,
+          WINHTTP_QUERY_STATUS_TEXT,
+          WINHTTP_HEADER_NAME_BY_INDEX,
+          outputBuffer.data(),
+          &sizeOfReasonPhrase,
+          WINHTTP_NO_HEADER_INDEX))
+  {
+    start = outputBuffer.begin();
+    reasonPhrase
+        = WideStringToString(std::wstring(start, start + sizeOfReasonPhrase / sizeof(WCHAR)));
+  }
+
+  // Allocate the instance of the response on the heap with a shared ptr so this memory gets
+  // delegated outside the transport and will be eventually released.
+  auto rawResponse
+      = std::make_unique<RawResponse>(majorVersion, minorVersion, httpStatusCode, reasonPhrase);
+
+  AddHeaders(responseHeaders, rawResponse);
+
+  int64_t contentLength
+      = GetContentLength(handleManager, requestMethod, rawResponse->GetStatusCode());
+
+  rawResponse->SetBodyStream(std::move(GetBodyStream(handleManager, contentLength)));
+  return rawResponse;
+}
+
+std::unique_ptr<RawResponse> WinHttpTransport::Send(Context const& context, Request& request)
+{
+  auto handleManager = std::make_unique<Details::HandleManager>(context, request);
+
+  CreateSessionHandle(handleManager);
+  CreateConnectionHandle(handleManager);
+  CreateRequestHandle(handleManager);
+
+  SendRequest(handleManager);
+
+  ReceiveResponse(handleManager);
+
+  return GetRawResponse(handleManager, request.GetMethod());
```

Comment:
```suggestion
  return GetRawResponse(std::move(handleManager), request.GetMethod());
```
GetRawResponse method must be updated to take the ownership of the handleManager. It will then transfer the ownership to the bodyStream inside the rawResponse.